src/api/Controllers/BoreholeGeometryController.cs

```diff
@@ -115,4 +111,42 @@ public async Task<IActionResult> UploadBoreholeGeometry(int boreholeId, IFormFil
             return Problem(errorMessage);
         }
     }
+
+    [HttpGet("[action]")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public async Task<IActionResult> GetDepthTVD([FromQuery] int boreholeId, [FromQuery] double depthMD)
+    {
+        var geometry = await GetBoreholeGeometry(boreholeId).ConfigureAwait(false);
+
+        if (geometry.Count < 2)
+        {
+            if (depthMD >= 0)
+            {
+                // Return the depthMD unchanged as if the borehole is perfectly vertical and infinitely long.
+                return Ok(depthMD);
+            }
+        }
+        else
+        {
+            try
+            {
+                return Ok(geometry.GetDepthTVD(depthMD));
+            }
+            catch (ArgumentOutOfRangeException)
+            {
```

Comment:
@patrickackermann  kannst du ab hier nochmal kurz übernehmen?

Reply:
@MiraGeowerkstatt Ich glaube die Begründung befindet sich ein paar Zeilen weiter unten im Kommentar `// Return an empty response in case the input was invalid`. Kannst du einfach das übernehmen? Danke :)